metricflow/model/parsing/dir_to_model.py

```diff
@@ -196,48 +219,47 @@ def parse_config_yaml(
+            try:
+                if document_type == METRIC_TYPE:
+                    metric_validator.validate(config_document[document_type])
+                    results.append(metric_class.parse_obj(object_cfg))
+                elif document_type == DATA_SOURCE_TYPE:
+                    data_source_validator.validate(config_document[document_type])
+                    results.append(data_source_class.parse_obj(object_cfg))
+                elif document_type == MATERIALIZATION_TYPE:
+                    materialization_validator.validate(config_document[document_type])
+                    results.append(materialization_class.parse_obj(object_cfg))
+                else:
+                    issues.append(
+                        ValidationError(
+                            context=FileContext(file_name=ctx.filename, line_number=ctx.start_line),
                             message=f"Invalid document type: {document_type}. Expected {DOCUMENT_TYPES}.",
-                            ctx=ctx,
-                            config_filepath=config_yaml.filepath,
                         )
                     )
+            # catches exceptions from jsonschema validator
+            except exceptions.ValidationError as e:
+                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
+                issues.append(
+                    ValidationError(
+                        context=context,
+                        message=f"YAML document did not conform to metric spec.\nError: {e}",
+                    )
                 )
-
-        if len(errors) > 0:
-            errors_str = "\n".join([str(x) for x in errors])
-            raise ParsingException(
-                message=f"Found {len(errors)} error(s) parsing configs files:\n"
-                f"{textwrap.indent(errors_str, prefix='    ')}"
-            )
-    except exceptions.ValidationError as e:
-        raise ParsingException(
-            message=f"YAML file did not conform to metric spec.\nError: {e}",
-            ctx=ctx,
-            config_filepath=config_yaml.filepath,
-        ) from e
-    except ScannerError as e:
-        raise ParsingException(
-            message=str(e),
-            ctx=ctx,
-            config_filepath=config_yaml.filepath,
-        ) from e
-    except ParsingException:
+            # catches exceptions from *.parse_obj calls
+            except ParsingException as e:
+                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
+                issues.append(ValidationError(context=context, message=str(e)))
+            # general exception for a given document. Basicially we don't want an exception
+            # on one document to halt checking the rest of the documents
+            except Exception as e:
+                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
+                issues.append(ValidationError(context=context, message=str(e)))
```

Comment:
These are identical, let's consolidate them.
```suggestion
            except (ParsingException, Exception) as e:
                # convert errors with *.parse_obj or general exceptions for a given document into issues
                context = FileContext(file_name=ctx.filename, line_number=ctx.start_line)
                issues.append(ValidationError(context=context, message=str(e)))
```